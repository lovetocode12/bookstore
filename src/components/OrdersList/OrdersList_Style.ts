import styled from "styled-components";

export const ContainerDiv=styled.div`
display:flex;
flex-direction:column;

.order_container{
    border: 1px solid #eee;
    margin-bottom: 1em;
    margin-left: 1em;
    padding: 0.3em 0.3em;
}

.order_status{
    justify-content: space-between;
    background-color: #e2c7c73b;
    padding: 1em;
    flex-wrap: wrap;
}

.order_status>div{
    width: 12em;
    white-space: nowrap;
}

.book_details{
    min-height: 5em;
    justify-content: space-evenly;
    font-size: 0.7em;
    flex-wrap: wrap;
    margin-left: 1em;
}

img{
    width: 10em;
}
`