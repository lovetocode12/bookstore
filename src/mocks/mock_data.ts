import { Book } from "../store/books/type";
import { Address, Order } from "../store/orders/type";

export class MockData {
    static Address: Address = {
        address1: 'address line 1',
        address2: 'address line 2',
        email: 'test@g.com',
        city: 'city',
        phone: '12345'
    }

    static Order: Order = {
        orderDate: new Date(),
        orderStatus: 'InProcess',
        book: {
            "title": "Unlocking Android",
            "id": "1933988673",
            "price": 25,
            "pageCount": 416,
            "url": "https://s3.amazonaws.com/AKIAJC5RLADLUMVRPFDQ.book-thumb-images/ableson.jpg",
            "description": "Unlocking Android: A Developer's Guide provides concise, hands-on instruction for the Android operating system and development tools. This book teaches important architectural concepts in a straightforward writing style and builds on this with practical and useful examples throughout.",
            "author": "W. Frank Ableson"
        }
    }
    static books: Book[] = [
        {
            "title": "Unlocking Android",
            "id": "1933988673",
            "price": 25,
            "pageCount": 416,
            "url": "https://s3.amazonaws.com/AKIAJC5RLADLUMVRPFDQ.book-thumb-images/ableson.jpg",
            "description": "Unlocking Android: A Developer's Guide provides concise, hands-on instruction for the Android operating system and development tools. This book teaches important architectural concepts in a straightforward writing style and builds on this with practical and useful examples throughout.",
            "author": "W. Frank Ableson"
        },
        {
            "title": "Android in Action, Second Edition",
            "id": "1935182722",
            "price": 22,
            "pageCount": 592,
            "url": "https://s3.amazonaws.com/AKIAJC5RLADLUMVRPFDQ.book-thumb-images/ableson2.jpg",
            "description": "Android in Action, Second Edition is a comprehensive tutorial for Android developers. Taking you far beyond \"Hello Android,\" this fast-paced book puts you in the driver's seat as you learn important architectural concepts and implementation strategies. You'll master the SDK, build WebKit apps using HTML 5, and even learn to extend or replace Android's built-in features by building useful and intriguing examples. ",
            "author": "W. Frank Ableson"
        },
        {
            "title": "Zend Framework in Action",
            "id": "1933988320",
            "price": 22,
            "pageCount": 432,
            "url": "https://s3.amazonaws.com/AKIAJC5RLADLUMVRPFDQ.book-thumb-images/allen.jpg",
            "description": "Zend Framework in Action is a comprehensive tutorial that shows how to use the Zend Framework to create web-based applications and web services. This book takes you on an over-the-shoulder tour of the components of the Zend Framework as you build a high quality, real-world web application.",
            "author": "Rob Allen"
        },
        {
            "title": "Flex on Java",
            "id": "1933988797",
            "pageCount": 265,
            "price": 22,
            "url": "https://s3.amazonaws.com/AKIAJC5RLADLUMVRPFDQ.book-thumb-images/allmon.jpg",
            "description": "   A beautifully written book that is a must have for every Java Developer.       Ashish Kulkarni, Technical Director, E-Business Software Solutions Ltd.",
            "author": "Bernerd Allmon"
        },
        {
            "title": "Griffon in Action",
            "id": "1935182234",
            "price": 22,
            "pageCount": 375,
            "url": "https://s3.amazonaws.com/AKIAJC5RLADLUMVRPFDQ.book-thumb-images/almiray.jpg",
            "description": "Griffon in Action is a comprehensive tutorial written for Java developers who want a more productive approach to UI development. In this book, you'll immediately dive into Griffon. After a Griffon orientation and a quick Groovy tutorial, you'll start building examples that explore Griffon's high productivity approach to Swing development. One of the troublesome parts of Swing development is the amount of Java code that is required to get a simple application off the ground.",
            "author": "Andres Almiray"
        },
        {
            "title": "OSGi in Depth",
            "id": "193518217X",
            "price": 22,
            "pageCount": 325,
            "url": "https://s3.amazonaws.com/AKIAJC5RLADLUMVRPFDQ.book-thumb-images/alves.jpg",
            "description": "Enterprise OSGi shows a Java developer how to develop to the OSGi Service Platform Enterprise specification, an emerging Java-based technology for developing modular enterprise applications. Enterprise OSGi addresses several shortcomings of existing enterprise platforms, such as allowing the creation of better maintainable and extensible applications, and provide a simpler, easier-to-use, light-weight solution to enterprise software development.",
            "author": "Alexandre de Castro Alves"
        },
        {
            "title": "Flexible Rails",
            "id": "1933988509",
            "price": 22,
            "pageCount": 592,
            "url": "https://s3.amazonaws.com/AKIAJC5RLADLUMVRPFDQ.book-thumb-images/alves.jpg",
            "description": "Enterprise OSGi shows a Java developer how to develop to the OSGi Service Platform Enterprise specification, an emerging Java-based technology for developing modular enterprise applications. Enterprise OSGi addresses several shortcomings of existing enterprise platforms, such as allowing the creation of better maintainable and extensible applications, and provide a simpler, easier-to-use, light-weight solution to enterprise software development.",
            "author": "Alexandre de Castro Alves"
        }
    ]

}