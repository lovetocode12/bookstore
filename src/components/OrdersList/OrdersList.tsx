import React, { useContext, useEffect } from 'react'
import { Order } from '../../store/orders/type'
import { useSelector } from 'react-redux';
import { ContainerDiv } from './OrdersList_Style';
import { TitleContext } from '../../common/TitleContext';
import { RootState } from '../../store/type';

type Props = {};

const OrdersList: React.FC<Props> = () => {
    
    const orders: any = useSelector((state: RootState) => state.orderState.pastOrders);
    const context: any = useContext(TitleContext);
    const dispatchTitle = context.dispatchTitle;
    useEffect(() => {
        if (dispatchTitle) {
            dispatchTitle({ type: 'SET_TITLE', payload: 'ORDERS' })
        }
    }, [dispatchTitle])

    return (
        <ContainerDiv>
            {
                orders.length > 0 ? orders.map((order: Order, index: number) =>
                    <div key={index} className="flex-col order_container">
                        <div className="flex-row order_status">
                            <div>Order Placed : {new Date(order.orderDate).toDateString().split(' ').slice(1).join(' ')}</div>
                            <div>Status : {order.orderStatus}</div>
                        </div>
                        <div className="flex-row flex-wrap">
                            <div>
                                <img src={order.book?.url} alt={order.book?.title} />
                            </div>
                            <div className="flex-col book_details">
                                <div><span className="mediumHeading">{order.book?.title}</span></div>
                                <div>Author Name  <span className="author_container">{order.book?.author}</span></div>
                                <div>Book Price <span className="price">{order.book?.price}</span></div>
                            </div>
                        </div>
                    </div>

                ) : "No Orders"
            }
        </ContainerDiv>
    )
}
export default OrdersList;

