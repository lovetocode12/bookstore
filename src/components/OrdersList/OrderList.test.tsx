import React from 'react';
import { shallow } from '../../../enzyme';
import { MockData } from '../../mocks/mock_data';
import OrdersList from './OrdersList';

jest.mock('react-redux', () => {
    const ActualReactRedux = require.requireActual('react-redux');
    return {
        ...ActualReactRedux,
        useSelector: jest.fn().mockImplementation(() => {
            return MockData.books;
        }),
        useDispatch: jest.fn().mockImplementation(() => {
            return [];
        }),
    };
});

describe('Orders List Component', () => {

    function getWrapper() {
        const wrapper: any = shallow(<OrdersList />);
        return wrapper;
    }

    it('Component should render correctly', () => {
        const wrapper = getWrapper();
        expect(wrapper).toMatchSnapshot();
    })
});