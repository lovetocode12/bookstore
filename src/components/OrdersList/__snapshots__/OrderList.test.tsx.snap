// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders List Component Component should render correctly 1`] = `
<styled.div>
  <div
    className="flex-col order_container"
    key="0"
  >
    <div
      className="flex-row order_status"
    >
      <div>
        Order Placed : 
        Date
      </div>
      <div>
        Status : 
      </div>
    </div>
    <div
      className="flex-row flex-wrap"
    >
      <div>
        <img />
      </div>
      <div
        className="flex-col book_details"
      >
        <div>
          <span
            className="mediumHeading"
          />
        </div>
        <div>
          Author Name  
          <span
            className="author_container"
          />
        </div>
        <div>
          Book Price 
          <span
            className="price"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex-col order_container"
    key="1"
  >
    <div
      className="flex-row order_status"
    >
      <div>
        Order Placed : 
        Date
      </div>
      <div>
        Status : 
      </div>
    </div>
    <div
      className="flex-row flex-wrap"
    >
      <div>
        <img />
      </div>
      <div
        className="flex-col book_details"
      >
        <div>
          <span
            className="mediumHeading"
          />
        </div>
        <div>
          Author Name  
          <span
            className="author_container"
          />
        </div>
        <div>
          Book Price 
          <span
            className="price"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex-col order_container"
    key="2"
  >
    <div
      className="flex-row order_status"
    >
      <div>
        Order Placed : 
        Date
      </div>
      <div>
        Status : 
      </div>
    </div>
    <div
      className="flex-row flex-wrap"
    >
      <div>
        <img />
      </div>
      <div
        className="flex-col book_details"
      >
        <div>
          <span
            className="mediumHeading"
          />
        </div>
        <div>
          Author Name  
          <span
            className="author_container"
          />
        </div>
        <div>
          Book Price 
          <span
            className="price"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex-col order_container"
    key="3"
  >
    <div
      className="flex-row order_status"
    >
      <div>
        Order Placed : 
        Date
      </div>
      <div>
        Status : 
      </div>
    </div>
    <div
      className="flex-row flex-wrap"
    >
      <div>
        <img />
      </div>
      <div
        className="flex-col book_details"
      >
        <div>
          <span
            className="mediumHeading"
          />
        </div>
        <div>
          Author Name  
          <span
            className="author_container"
          />
        </div>
        <div>
          Book Price 
          <span
            className="price"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex-col order_container"
    key="4"
  >
    <div
      className="flex-row order_status"
    >
      <div>
        Order Placed : 
        Date
      </div>
      <div>
        Status : 
      </div>
    </div>
    <div
      className="flex-row flex-wrap"
    >
      <div>
        <img />
      </div>
      <div
        className="flex-col book_details"
      >
        <div>
          <span
            className="mediumHeading"
          />
        </div>
        <div>
          Author Name  
          <span
            className="author_container"
          />
        </div>
        <div>
          Book Price 
          <span
            className="price"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex-col order_container"
    key="5"
  >
    <div
      className="flex-row order_status"
    >
      <div>
        Order Placed : 
        Date
      </div>
      <div>
        Status : 
      </div>
    </div>
    <div
      className="flex-row flex-wrap"
    >
      <div>
        <img />
      </div>
      <div
        className="flex-col book_details"
      >
        <div>
          <span
            className="mediumHeading"
          />
        </div>
        <div>
          Author Name  
          <span
            className="author_container"
          />
        </div>
        <div>
          Book Price 
          <span
            className="price"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex-col order_container"
    key="6"
  >
    <div
      className="flex-row order_status"
    >
      <div>
        Order Placed : 
        Date
      </div>
      <div>
        Status : 
      </div>
    </div>
    <div
      className="flex-row flex-wrap"
    >
      <div>
        <img />
      </div>
      <div
        className="flex-col book_details"
      >
        <div>
          <span
            className="mediumHeading"
          />
        </div>
        <div>
          Author Name  
          <span
            className="author_container"
          />
        </div>
        <div>
          Book Price 
          <span
            className="price"
          />
        </div>
      </div>
    </div>
  </div>
</styled.div>
`;
